persistence/src/main/java/edu/unc/lib/dl/persist/services/move/MoveObjectsJob.java

```diff
@@ -0,0 +1,317 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.persist.services.move;
+
+import static edu.unc.lib.dl.fcrepo4.RepositoryPathConstants.FCR_TOMBSTONE;
+
+import java.io.IOException;
+import java.net.URI;
+import java.security.SecureRandom;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.jena.query.QueryExecution;
+import org.apache.jena.query.QuerySolution;
+import org.apache.jena.query.ResultSet;
+import org.apache.jena.rdf.model.Resource;
+import org.fcrepo.client.FcrepoClient;
+import org.fcrepo.client.FcrepoOperationFailedException;
+import org.fcrepo.client.FcrepoResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.unc.lib.dl.acl.service.AccessControlService;
+import edu.unc.lib.dl.acl.util.AgentPrincipals;
+import edu.unc.lib.dl.acl.util.Permission;
+import edu.unc.lib.dl.fcrepo4.ContentContainerObject;
+import edu.unc.lib.dl.fcrepo4.ContentObject;
+import edu.unc.lib.dl.fcrepo4.FedoraTransaction;
+import edu.unc.lib.dl.fcrepo4.PIDs;
+import edu.unc.lib.dl.fcrepo4.RepositoryObject;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectLoader;
+import edu.unc.lib.dl.fcrepo4.TransactionManager;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.fedora.ServiceException;
+import edu.unc.lib.dl.reporting.ActivityMetricsClient;
+import edu.unc.lib.dl.search.solr.model.ObjectPath;
+import edu.unc.lib.dl.search.solr.service.ObjectPathFactory;
+import edu.unc.lib.dl.services.OperationsMessageSender;
+import edu.unc.lib.dl.sparql.SparqlQueryService;
+import edu.unc.lib.dl.util.DateTimeUtil;
+
+/**
+ * Job which performs a single move operation to transfer a list of objects from
+ * their source containers to a single destination content container
+ *
+ * @author bbpennel
+ *
+ */
+public class MoveObjectsJob implements Runnable {
+    private static final Logger log = LoggerFactory.getLogger(MoveObjectsJob.class);
+    private static final Logger moveLog = LoggerFactory.getLogger("move_logger");
+
+    private AccessControlService aclService;
+    private RepositoryObjectLoader repositoryObjectLoader;
+    private TransactionManager transactionManager;
+    private SparqlQueryService sparqlQueryService;
+    private FcrepoClient fcrepoClient;
+    private OperationsMessageSender operationsMessageSender;
+    private ObjectPathFactory objectPathFactory;
+    private ActivityMetricsClient operationMetrics;
+
+    private AgentPrincipals agent;
+    private PID destination;
```

Comment:
would you mind renaming this "destinationPid" for consistency and clarity?

Reply:
renamed